arealite/README.md

```diff
@@ -0,0 +1,213 @@
+
+AReaLite is the first step in AReaL's refactoring process. It is not only a standalone training library with shallow interfaces, but will also provide the core API definitions to be used by AReaL in the future. AReaL will essentially transform its current worker-based architecture into an AI-centric architecture like AReaLite. AReaL will **extend** AReaLite's APIs and implementations to support more backends for efficient large-scale training.
+
+## Expectations of AReaLite
+
+### Highlights
+
++ Fully asynchronous training with decoupled inference and training.
++ Elastic inference device scaling — users can shut down or launch more inference processes independently during training.
++ Full SFT/RL algorithmic functionality matching AReaL.
++ Arbitrary agentic rollout collector customization in a single file.
++ Easy navigation to implementation references via Ctrl+click in VSCode.
++ Support for distributed launching with Ray/SLURM/torchrun.
+
+### AReaLite's Scope
+
++ Not bound to Ray.
++ Only supports SGLang and PyTorch FSDP2 with SPMD launching.
++ No customized data structures like `SequenceSample`. All data are PyTorch tensors.
++ Uses HuggingFace (models, datasets) and PyTorch (FSDP, data structures) as much as possible.
+
+## Architecture
+
+### Core Components
+
+```
+arealite/
+├── api/           # Abstract interfaces and data structures
+├── impl/          # Concrete implementations
+├── cli/           # Command-line interfaces
+├── config/        # Configuration templates
+└── tests/         # Standalone test scripts
+```
+
+#### 1. API Layer (`api/`)
+
+The API layer defines abstract interfaces and data structures that provide a clean contract between different components:
+
+- **`engine_api.py`**: Defines `SPMDWrapper` for SPMD-based training backends (FSDP) and `EngineFactory`
+- **`trainer_api.py`**: Defines `Trainer` base class for different training algorithms and `TrainerFactory`
+- **`rollout_api.py`**: Defines `RolloutCollector`, `Agent`, `Environment` for RL data collection and `RolloutCollectorFactory`
+- **`cli_args.py`**: Defines configuration dataclasses for all components
+
+#### 2. Implementation Layer (`impl/`)
+
+The implementation layer contains concrete implementations of the API interfaces:
+
+- **`fsdp_wrapper.py`**: FSDP-based training engine using PyTorch FSDP2
+- **`trainer/grpo.py`**: GRPO trainer implementation for reinforcement learning
+- **`rollout_controller.py`**: Coordinates rollout data collection across workers
+- **`rlvr/`**: RLVR (RL via Verification and Refinement) collector implementations
```

Comment:
Shouldn't it stands for `Reinforcement Learning with Verifiable Rewards` as we commonly used?